### PR TITLE
Support Precogs

### PR DESCRIPTION
**Pull request by vectrixdevelops**
Precogs is an AntiCheat service API that we will support. Meaning if you're plugins conflict with the AntiCheat you will be able to control detections through it's lightweight API. This isn't quite ready to be merged yet, but is coming.

### CODE REVIEW

File: src/main/java/io/github/connorhartley/guardian/Guardian.java [hunk 1]
```diff
@@ -62,7 +66,10 @@
         description = "An extensible anticheat plugin for Sponge.",
         authors = {
                 "Connor Hartley (vectrix)"
-        }
+        },
+        dependencies = @Dependency(
```
Comment (vectrixdevelops) [inline src/main/java/io/github/connorhartley/guardian/Guardian.java:71]: This should be an optional dependency.

**Review by vectrixdevelops — COMMENTED** [review]
Just some small changes required.

File: src/main/java/io/github/connorhartley/guardian/Guardian.java (hunk 1 above)
Comment (me4502) [reply src/main/java/io/github/connorhartley/guardian/Guardian.java:71]: The way it's setup it needs to be there. It'd require a fair few changes to be optional.

Comment (vectrixdevelops) [reply src/main/java/io/github/connorhartley/guardian/Guardian.java:71]: Ah yeah true, I underestimated that.

**Pull request merged**